### PR TITLE
Fix test_worktree_add_rm test for Windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Fixes the `test_worktree_add_rm` test in `modules.test_git`. The test was being skipped because the `_worktrees_supported` function was always getting an `AttributeError` due to the `.encode()`. This was a problem in Linux as well.

Fixed the actual test for Windows. The problem is that Git returns Unix style paths. Had to change the Windows paths to Unix style paths for comparison purposes.

Removed the `worktree_basename2` variable as it wasn't being used in the test.

### Tests written?
Yes

### CODE REVIEW

**Comment by twangboy** [comment]
Go Go Jenkins!

File: tests/integration/modules/test_git.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def _worktrees_supported():
     Check if the git version is 2.5.0 or later
     '''
     try:
-        return _git_version() >= LooseVersion('2.5.0').encode()
+        return _git_version() >= LooseVersion('2.5.0')
```
Comment (cachedout) [inline tests/integration/modules/test_git.py]: A git history shows that this `.encode()` was added for Py3 compat. Does this change regress Py3 support?

File: tests/integration/modules/test_git.py [hunk 2]
```diff
@@ -931,10 +931,16 @@ def test_worktree_add_rm(self):
...
+
         worktree_path = tempfile.mkdtemp(dir=integration.TMP)
         worktree_basename = os.path.basename(worktree_path)
         worktree_path2 = tempfile.mkdtemp(dir=integration.TMP)
-        worktree_basename2 = os.path.basename(worktree_path2)
+
+        # Even though this is Windows, git commands return a unix style path
+        if salt.utils.is_windows():
+            worktree_path = worktree_path.replace('\\', '/')
+            worktree_path = worktree_path.replace('\\', '/')
```
Comment (cachedout) [inline tests/integration/modules/test_git.py]: Two questions. First, why is this being run twice? Second, if these are being re-written to UNIX-style paths, how is the check below for `is_worktree` passing on Windows? Shouldn't that fail if you pass it UNIX-style paths on Windows?

File: tests/integration/modules/test_git.py (hunk 1 above)
Comment (twangboy) [inline tests/integration/modules/test_git.py]: So, how do I make it work in both?

File: tests/integration/modules/test_git.py (hunk 2 above)
Comment (twangboy) [inline tests/integration/modules/test_git.py]: 1. This was a typo on my part, should be `worktree_path2`
2. The output of Git returns Unix style os.sep on Windows

I will fix #1

File: tests/integration/modules/test_git.py (hunk 1 above)
Comment (twangboy) [inline tests/integration/modules/test_git.py]: `_git_version()` returns `LooseVersion()` without the `.encode()`
The `LooseVersion()` class doesn't have an `encode()` method. Perhaps they meant to put it inside the parenthesis... (`LooseVersion('2.5.0'.encode())` I'll try that...

Comment (twangboy) [inline tests/integration/modules/test_git.py]: Yeah, that worked. Updated PR.

**Pull request merged**